### ISSUE TITLE
Unable to extract nsig function code 

### ISSUE THREAD

**Issue by yeroca** [ytdl-org/youtube-dl#32970]
## Checklist

- [ ] I'm reporting a broken site support issue
- [X] I've verified that I'm running youtube-dl version **2021.12.17**
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [X] I've searched the bugtracker for similar bug reports including closed ones
- [X] I've read bugs section in FAQ

## Verbose log

```
% ./youtube-dl -v 'https://youtube.com/shorts/Z4js1seup0o?si=fGTXOUdYnNas1xlw'
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://youtube.com/shorts/Z4js1seup0o?si=fGTXOUdYnNas1xlw']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Single file build
[debug] Python 3.13.0 (CPython x86_64 64bit) - Linux-6.11.5-300.fc41.x86_64-x86_64-with-glibc2.40 - OpenSSL 3.2.2 4 Jun 2024 - glibc 2.40
[debug] exe versions: ffmpeg 7.0.2, ffprobe 7.0.2
[debug] Proxy map: {}
[youtube] Z4js1seup0o: Downloading webpage
[youtube] Z4js1seup0o: Downloading player baafab19
ERROR: Unable to extract nsig function code (caused by RegexNotFoundError('Unable to extract \x1b[0;34mInitial JS player n function name\x1b[0m; please report this issue on https://github.com/ytdl-org/youtube-dl/issues , using the appropriate issue template. Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose option and include the complete output.')); please report this issue on https://github.com/ytdl-org/youtube-dl/issues , using the appropriate issue template. Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose option and include the complete output.
Traceback (most recent call last):
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1637, in _decrypt_nsig
    jsi, player_id, func_code = self._extract_n_function_code(video_id, player_url)
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1686, in _extract_n_function_code
    func_name = self._extract_n_function_name(jscode)
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1660, in _extract_n_function_name
    func_name, idx = self._search_regex(
                     ~~~~~~~~~~~~~~~~~~^
        # new: (b=String.fromCharCode(110),c=a.get(b))&&c=nfunc[idx](c)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<5 lines>...
            (?P<nfunc>[a-zA-Z_$][\w$]*)(?:\s*\[(?P<idx>\d+)\])?\s*\(\s*[\w$]+\s*\)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        ''', jscode, 'Initial JS player n function name', group=('nfunc', 'idx'))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/common.py", line 1101, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://github.com/ytdl-org/youtube-dl/issues , using the appropriate issue template. Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose option and include the complete output.
Traceback (most recent call last):
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1637, in _decrypt_nsig
    jsi, player_id, func_code = self._extract_n_function_code(video_id, player_url)
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1686, in _extract_n_function_code
    func_name = self._extract_n_function_name(jscode)
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1660, in _extract_n_function_name
    func_name, idx = self._search_regex(
                     ~~~~~~~~~~~~~~~~~~^
        # new: (b=String.fromCharCode(110),c=a.get(b))&&c=nfunc[idx](c)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<5 lines>...
            (?P<nfunc>[a-zA-Z_$][\w$]*)(?:\s*\[(?P<idx>\d+)\])?\s*\(\s*[\w$]+\s*\)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        ''', jscode, 'Initial JS player n function name', group=('nfunc', 'idx'))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/common.py", line 1101, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://github.com/ytdl-org/youtube-dl/issues , using the appropriate issue template. Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose option and include the complete output.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/YoutubeDL.py", line 875, in wrapper
    return func(self, *args, **kwargs)
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/YoutubeDL.py", line 971, in __extract_info
    ie_result = ie.extract(url)
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/common.py", line 571, in extract
    ie_result = self._real_extract(url)
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 2104, in _real_extract
    self._unthrottle_format_urls(video_id, player_url, dct)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1721, in _unthrottle_format_urls
    n_response = decrypt_nsig(n_param)(n_param, video_id, player_url)
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1614, in inner
    raise ret
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1606, in inner
    self._player_cache[cache_id] = func(*args, **kwargs)
                                   ~~~~^^^^^^^^^^^^^^^^^
  File "/home/corey/git-trees/youtube-dl/./youtube-dl/youtube_dl/extractor/youtube.py", line 1639, in _decrypt_nsig
    raise ExtractorError('Unable to extract nsig function code', cause=e)
youtube_dl.utils.ExtractorError: Unable to extract nsig function code (caused by RegexNotFoundError('Unable to extract \x1b[0;34mInitial JS player n function name\x1b[0m; please report this issue on https://github.com/ytdl-org/youtube-dl/issues , using the appropriate issue template. Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose option and include the complete output.')); please report this issue on https://github.com/ytdl-org/youtube-dl/issues , using the appropriate issue template. Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose option and include the complete output.

```

## Description

In the last couple of months, I've tried to download a youtube videos on a several occasions, and ran into this error.  At least I think it was this same error.  It doesn't seem to matter which video I try to download.

**Comment by dirkf** [comment]
Please review #30839 to check whether your version is expected to work with YT (if it's actually 2021.12.17 it isn't) and #32905 regarding work-arounds temporarily required with the latest nightly version.

**Comment by yeroca** [comment]
It seems I made an incorrect assumption that cloning the git tree from this repo and checking out the master would give me the latest stable code.

I just now tried the nightly build and it worked fine.  Sorry for opening this issue.

**Comment by dirkf** [comment]
The master branches in the main and nightly repos should be the same apart from code that is specific to the nightly build (naming, etc), except for a few hours after a commit in the main repo, or if the sync action has to be disabled temporarily (it hasn't).

If you checked out the 2021.12.17 tag from the master repo rather than the latest version, you would indeed get the logged behaviour. The nightly repo is a better target because it shows accurate version details.

**Comment by yeroca** [comment]
Thanks for the clarifications!

As far as I remember, I just did a straight-up clone, and didn't specify a branch or tag.  Each time I tried it and ran into an error, I did a pull, and it kept saying my tree is up to date.  I suppose it's possible I tried checking out a particular release, thinking the top of the tree might be broken, and then forgot that I had done that later. In any case, I deleted the entire tree so there's no way to know anymore.  I'm currently tracking the nightly repo.
